From 246076c53f009c55fbf320e665df9648ac3f9c44 Mon Sep 17 00:00:00 2001
From: ArenM <aren@peacevolution.org>
Date: Wed, 4 May 2022 23:51:15 -0400
Subject: [PATCH sxmo-utils 5/5] fix conflicting audio settings (#502)

When callaudiod disables call audio mode it always enables the mic.
Because we were disabling the mic before disabling call audio mode
reset_audio always fails.

I also removed the retry loop; if the software we rely on breaks we
should fix it upstream, rather than hitting it with a hammer until it
works.

This should fix https://todo.sr.ht/~mil/sxmo-tickets/502
---
 scripts/modem/sxmo_modemaudio.sh | 42 +++++++++++---------------------
 1 file changed, 14 insertions(+), 28 deletions(-)

diff --git a/scripts/modem/sxmo_modemaudio.sh b/scripts/modem/sxmo_modemaudio.sh
index 1bdfedc..8e02ad8 100644
--- a/scripts/modem/sxmo_modemaudio.sh
+++ b/scripts/modem/sxmo_modemaudio.sh
@@ -3,44 +3,30 @@
 # shellcheck source=scripts/core/sxmo_common.sh
 . sxmo_common.sh
 
+set -e
+
 setup_audio() {
-	unmute_mic
 	enable_call_audio_mode
 	disable_speaker
+	unmute_mic
 
-	i=0
-	while (! is_unmuted_mic) || (! is_call_audio_mode) || (! is_disabled_speaker); do
-		i=$((i+1))
-		if [ "$i" -gt 5 ]; then
-			return 1
-		fi
-
-		sleep 0.2
-
-		unmute_mic
-		enable_call_audio_mode
-		disable_speaker
-	done
+	# TODO: callaudiocli should error if it doesn't set these
+	# Since we set -e this script will error if these aren't true
+	is_call_audio_mode
+	is_disabled_speaker
+	is_unmuted_mic
 }
 
 reset_audio() {
-	mute_mic
 	disable_call_audio_mode
 	enable_speaker
+	mute_mic
 
-	i=0
-	while (! is_muted_mic) || (! is_default_audio_mode) || (! is_enabled_speaker); do
-		i=$((i+1))
-		if [ "$i" -gt 5 ]; then
-			return 1
-		fi
-
-		sleep 0.2
-
-		mute_mic
-		disable_call_audio_mode
-		enable_speaker
-	done
+	# TODO: callaudiocli should error if it doesn't set these
+	# Since we set -e this script will error if these aren't true
+	is_default_audio_mode
+	is_enabled_speaker
+	is_muted_mic
 }
 
 is_muted_mic() {
-- 
2.36.1

